wgpu-core/src/device/mod.rs

```diff
@@ -3870,6 +3870,10 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                 );
                 config.present_mode = hal::window::PresentMode::FIFO;
             }
+            if width == 0 || height == 0 {
+                return Err(swap_chain::CreateSwapChainError::ZeroAreaSwapchain);
```

Comment:
We don't need the `Swapchain` suffix of the error variant of `CreateSwapChainError`